code/modules/clothing/clothing.dm

```diff
@@ -987,6 +987,45 @@
 		to_chat(usr, "<span class='notice'>You roll down your [src]'s sleeves.</span>")
 	update_clothing_icon()
 
+//Switch to taur sprites if a taur equips
+/obj/item/clothing/suit
+	var/taurized = FALSE
+
+/obj/item/clothing/suit/equipped(var/mob/user, var/slot)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if((taurized && !isTaurTail(H.tail_style)) || (!taurized && isTaurTail(H.tail_style)))
+			taurize(user)
+
+	return ..()
+
+/obj/item/clothing/suit/proc/taurize(var/mob/living/carbon/human/Taur)
+	if(isTaurTail(Taur.tail_style))
+		var/datum/sprite_accessory/tail/taur/taurtail = Taur.tail_style
+		if(taurtail.suit_sprites && (get_worn_icon_state(slot_wear_suit_str) in cached_icon_states(taurtail.suit_sprites)))
+			icon_override = taurtail.suit_sprites
+			taurized = TRUE
+
+	if(!taurized)
+		icon_override = initial(icon_override)
+		taurized = FALSE
+
+// Taur suits need to be shifted so its centered on their taur half.
+/obj/item/clothing/suit/make_worn_icon(var/body_type,var/slot_name,var/inhands,var/default_icon,var/default_layer = 0,var/icon/clip_mask)
+	var/image/standing = ..()
+	if(taurized) //Special snowflake var on suits
+		standing.pixel_x = -16
+		standing.layer = BODY_LAYER + 15 // 15 is above tail layer, so will not be covered by taurbody.
+	return standing
+
+/obj/item/clothing/suit/apply_accessories(var/image/standing)
+	if(LAZYLEN(accessories) && taurized)
+		for(var/obj/item/clothing/accessory/A in accessories)
+			var/image/I = new(A.get_mob_overlay())
+			I.pixel_x = 16 //Opposite of the pixel_x on the suit (-16) from taurization to cancel it out and puts the accessory in the correct place on the body.
+			standing.add_overlay(I)
+	else
+		return ..()

```

Comment:
Similarly, all of this should be moved up to be with the rest of /obj/item/clothing/suit's definitions